internal/web/server.go

```diff
@@ -216,29 +206,64 @@ func (s *server) installModalVersions(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprintf(os.Stderr, "An error occurred fetching versions of %v: %v\n", pkgName, err)
 		}
 	}
-
 	err := pkgInstallModalVersionsTmpl.Execute(w, &map[string]any{
 		"ShowVersions": showVersions,
 		"PackageIndex": &idx,
 	})
 	checkTmplError(err, "pkgInstallModalVersionsTmpl")
 }
 
+func (s *server) updateModal(w http.ResponseWriter, r *http.Request) {
+	pkgName := r.FormValue("packageName")
+	var pkg v1alpha1.Package
+	if err := s.pkgClient.Packages().Get(r.Context(), pkgName, &pkg); err != nil {
+		fmt.Fprintf(os.Stderr, "An error occurred fetching package %v: \n%v\n", pkgName, err)
+		return
+	}
+
+	if latestVersion, err := repo.GetLatestVersion("", pkgName); err != nil {
+		fmt.Fprintf(os.Stderr, "An error occurred fetching latest version of %v: \n%v\n", pkgName, err)
+	} else {
+		err := pkgUpdateModalTmpl.Execute(w, &map[string]any{
+			"Package":       pkg,
+			"LatestVersion": latestVersion,
+		})
+		checkTmplError(err, "pkgUpdateModalTmpl")
+	}
+}
+
+func (s *server) update(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	pkgName := r.FormValue("packageName")
+
+	updater := update.NewUpdater(s.pkgClient).WithStatusWriter(statuswriter.Stderr())
```

Comment:
The updater is actually designed to be reusable, so I think it would be better to have it as a server property.

Reply:
As discussed, we will leave it as it is for now, because we don't know how the status writer handles concurrent usage.